### PR TITLE
CDS-68067 custom trigger payload ex

### PR DESCRIPTION
**Pull request by michaelcretzman**
https://harness.atlassian.net/browse/CDS-68067

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ x] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-68067

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/11_Triggers/custom-trigger-passing-data.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+matrix:
+  emailId: <+pipeline.variables.emailIds>.split(',')
+maxConcurrency: 2
+```
+
+You can see the split() method is used to parse the email addresses in the JSON payload.
+
+In the **To** setting of the Email step, you can reference the matrix using `<+matrix.emaild>`.
+
+Now the Email step is run for each addresses in the matrix.
```
Comment (brian-f-harness) [inline docs/platform/11_Triggers/custom-trigger-passing-data.md:122]:
```suggestion
Now the Email step is run for each address in the matrix.
```

File: docs/platform/11_Triggers/custom-trigger-passing-data.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+For example, in the following cURL command, the key name is `my_key`:
+
+```
+curl -X POST -H 'content-type: application/json' --url 'https://app.harness.io/gateway/pipeline/api/webhook/custom/v2?accountIdentifier=H5W8iol5TNWc4G9h5A2MXg&orgIdentifier=default&projectIdentifier=CD_Docs&pipelineIdentifier=Custom&triggerIdentifier=Custom' -d '{"my_key": "hello world"}'
+```
+
+You would reference this data using the expression `<+trigger.payload.my_key>`. The expression would resolve to `hello world`.
+
+
+You can use the `<+trigger.payload.[key_name]>` expression is a script, a variable value, or any setting configured as an [expression](/docs/platform/20_References/runtime-inputs).
```
Comment (brian-f-harness) [inline docs/platform/11_Triggers/custom-trigger-passing-data.md:40]:
```suggestion
You can use the `<+trigger.payload.[key_name]>` expression as a script, a variable value, or any setting configured as an [expression](/docs/platform/20_References/runtime-inputs).
```

**Review by brian-f-harness — APPROVED** [review]
lgtm

File: docs/platform/11_Triggers/custom-trigger-passing-data.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+<docimage path={require('./static/trigger-deployments-using-custom-triggers-00.png')} width="60%" height="60%" title="Click to view full size image" />
+
+For information on creating Custom triggers, go to [Trigger deployments using custom triggers](trigger-deployments-using-custom-triggers).
+
+You can use the Custom trigger cURL command to initiate the trigger and pass in data. 
+
+The cURL command will look something like this:
+
+```
+curl -X POST -H 'content-type: application/json' -H 'X-Api-Key: sample_api_key' --url 'https://app.harness.io/gateway/pipeline/api/webhook/custom/v2?accountIdentifier=H5W8iol5TNWc4G9h5A2MXg&orgIdentifier=default&projectIdentifier=CD_Docs&pipelineIdentifier=Custom&triggerIdentifier=Custom' -d '{"sample_key": "sample_value"}'
```
Comment (sribalij) [inline docs/platform/11_Triggers/custom-trigger-passing-data.md:24]: Not sure if we need a reference to API key section for custom triggers.
https://developer.harness.io/docs/platform/triggers/trigger-deployments-using-custom-triggers/#custom-trigger-authorization-using-api-keys

**Review by thisrohangupta — APPROVED** [review]

**Pull request merged**